include/app_memory/mem_domain.h

```diff
@@ -127,7 +127,12 @@ extern void k_mem_domain_init(struct k_mem_domain *domain, uint8_t num_parts,
 /**
  * @brief Destroy a memory domain.
  *
- * Destroy a memory domain.
+ * Destroy a memory domain. All member threads will be re-assigned to the
+ * default memory domain.
+ *
+ * The default memory domain may not be destroyed.
```

Comment:
Is this saying that the default memory domain cannot be destroyed (with this function), like it is not permissible? Since the members are re-assigned to the default, it would not make sense to destroy itself. sorry, my brain tried to also read it as a potential to not be destroyed

Reply:
ah there it is! "cannot destroy default domain" (line 281)